File: src/org/symbolmath/ast/leaf/LeafElement.java
package org.symbolmath.ast.leaf;

import org.symbolmath.ast.ASTElement;

import java.util.List;

public abstract class LeafElement extends ASTElement {

  public final void replaceChild(ASTElement element, ASTElement newElement) {
    throw new UnsupportedOperationException("replaceChild() not supported for leaf elements.");
  }

  @Override
  public List<ASTElement> getChildren() {
    return NO_CHILDREN;
  }
}

File: src/main/java/org/symbolmath/ast/leaf/LeafElement.kt
package org.symbolmath.ast.leaf

import org.symbolmath.ast.ASTElement

abstract class LeafElement : ASTElement() {

    override val children: List<ASTElement>
        get() = listOf()

    override fun replaceChild(element: ASTElement, newElement: ASTElement) {
        throw UnsupportedOperationException("replaceChild() not supported for leaf elements.")
    }

    override fun negate(): ASTElement {
        TODO("Not yet implemented")
    }

}